src/main/java/org/opensearch/neuralsearch/processor/ScoreCombinationTechnique.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+/**
+ * Collection of techniques for score combination
+ */
+public enum ScoreCombinationTechnique {
+
+    /**
+     * Arithmetic mean method for combining scores.
+     * cscore = (score1 + score2 +...+ scoreN)/N
+     *
+     * Zero (0.0) scores are excluded from number of scores N
+     */
+    ARITHMETIC_MEAN {
+
+        @Override
+        public float combine(float[] scores) {
+            float combinedScore = 0.0f;
+            int count = 0;
+            for (float score : scores) {
+                if (score >= 0.0) {
```

Comment:
why we are ignoring the 0 scores herre?

Reply:
I think we discussed this approach offline, score 0.0 means there is not hit for that doc in one particular sub-query. It's typically for text based queries like "term" or "match", in such case we're boosting results of neural searches.  
I think both approaches may have their use cases, we can add "vanilla" arithmetic mean and keep this one, so user chooses which one fits better.